### PR TITLE
Pipeline2.0: low hanging fruits - move buffers' common params to struct sof_audio_buffer.

### PR DESCRIPTION
**Pull request by marcinszkudlinski**
This PR moves many of parameters that are common for all buffers to struct sof_audio_buffer

there are 4 left:
```
	struct comp_dev *source;	/* source component */
	struct comp_dev *sink;		/* sink component */

	/* lists */
	struct list_item source_list;	/* list in comp buffers */
	struct list_item sink_list;	/* list in comp buffers */
```
and will be moved to sof_audio_buffer in next PR

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

**Review by cujomalainey — APPROVED** [review]

File: src/include/sof/audio/audio_buffer.h [hunk 1]
```diff
@@ -173,6 +173,21 @@ static inline bool audio_buffer_is_shared(struct sof_audio_buffer *buffer)
 	return buffer->is_shared;
 }
 
+static inline bool audio_buffer_is_hw_params_configured(struct sof_audio_buffer *buffer)
```
Comment (lyakh) [inline src/include/sof/audio/audio_buffer.h:182]: **are** hw params configured? Or maybe just `audio_buffer_hw_params_configured()` or `audio_buffer_check_hw_params_configured()`

**Review by kv2019i — APPROVED** [review]
The is/are_hw_params seems rather minor.

**Comment by marcinszkudlinski** [comment]
there's a problem with CI loadable library test, checking

**Comment by marcinszkudlinski** [comment]
internal CI that was failing is green, jenkins did not start. Lately there was couple of similar problems in several PRs 

BlobNotFoundThe specified blob does not exist. RequestId:edb2f1ad-201e-0086-7fa7-105dd6000000 Time:2024-09-27T06:33:40.4681943Z

**Comment by kv2019i** [comment]
SOFCI TEST

**Pull request merged**